src/parquet/file/reader-internal.h

```diff
@@ -103,8 +103,8 @@ class SerializedFile : public ParquetFileReader::Contents {
   // This class does _not_ take ownership of the data source. You must manage its
   // lifetime separately
   static std::unique_ptr<ParquetFileReader::Contents> Open(
-      std::unique_ptr<RandomAccessSource> source,
-      ReaderProperties props = default_reader_properties());
+      const std::shared_ptr<RandomAccessSource>& source,
```

Comment:
I can change it back, but semantically, why would we require that the file reader have exclusive access to a resource? You can always create a wrapper object that wraps a shared_ptr to some other IO interface (e.g., we could have `std::unique_ptr<ArrowInputFile>`, wrapping a shared ptr to one of the Arrow IO implementations).